cpp/src/aggregation/aggregation.cu

```diff
@@ -15,9 +15,13 @@
  */
 
 #include <cudf/detail/aggregation/aggregation.cuh>
+#include <cudf/detail/aggregation/aggregation.hpp>
+#include <cudf/table/table_view.hpp>
 
 #include <rmm/cuda_stream_view.hpp>
 
+#include <vector>
```

Comment:
I think we can remove this as it's being included from `aggregation.cuh`. Also curious if we need it in the first place or is it added by clangd?

Reply:
Thanks for raising your concern. I added those headers manually since they are missed in the first place. Ideally, we should never rely on transitive inclusions and always include everything used (https://google.github.io/styleguide/cppguide.html#Include_What_You_Use). I vaguely recall we had a bullet point for this in our guideline doc but I couldn't find it anymore.

Reply:
Thanks for clarifying this.